arch/xtensa/include/xtensa-asm2-context.h

```diff
@@ -127,4 +134,15 @@
 			  BASE_SAVE_AREA_SIZE_THREADPTR))
 #endif
 
+#if XCHAL_HAVE_FP && defined(CONFIG_CPU_HAS_FPU) && defined(CONFIG_FPU_SHARING)
+#define BSA_FPU_OFF \
+			(BASE_SAVE_AREA_SIZE - \
+			 (BASE_SAVE_AREA_SIZE_COMMON + \
+			  BASE_SAVE_AREA_SIZE_LOOPS + \
+			  BASE_SAVE_AREA_SIZE_EXCCAUSE + \
+			  BASE_SAVE_AREA_SIZE_SCOMPARE + \
+			  BASE_SAVE_AREA_SIZE_THREADPTR + \
+			  BASE_SAVE_AREA_SIZE_FPU))
```

Comment:
Every time I see this code I cringe a little.  It hasn't scaled well, but this is indeed the scheme we've picked so there's nothing wrong with extending it in the intended way.  Could do with a rewrite though.